### PR TITLE
Fix PhpDocs

### PR DESCRIPTION
**Pull request by blankse**

### CODE REVIEW

File: bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php [hunk 1]
```diff
@@ -131,7 +132,7 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
     }
 
     /**
-     * @param $user
+     * @param User|null $user
```
Comment (dvesh3) [inline bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php]:
```suggestion
     * @param User $user
```
passing null has no effect so I would avoid that.

Comment (dvesh3) [inline bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php]:
```suggestion
     * @param User $user
```
passing null has no effect so I would avoid that. Better adjust the if check.

File: models/DataObject/ClassDefinition/Data/Block.php [hunk 2]
```diff
@@ -46,35 +46,35 @@ class Block extends Data implements CustomResourcePersistingInterface, ResourceP
      *
      * @var bool
      */
-    public $lazyLoading;
+    public $lazyLoading = false;
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Block.php]: BC Break? as now it is always `bool` as oppose to `null` on initialize before:

```
$block = new DataObject\ClassDefinition\Data\Block();
if ($block->getLazyLoading() === null) {
    //do something
}
```

File: models/DataObject/ClassDefinition/Data/FieldDefinitionEnrichmentInterface.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ interface FieldDefinitionEnrichmentInterface
      *
      * @param array $context
      *
-     * @return Data
+     * @return $this
      */
-    public function enrichFieldDefinition(/** array */ $context = []) /** : Data */;
+    public function enrichFieldDefinition(/** array */ $context = []) /** : static */;
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/FieldDefinitionEnrichmentInterface.php:32]:
```suggestion
    public function enrichFieldDefinition(/** array */ $context = []) /** : self */;
```

File: models/DataObject/ClassDefinition/Data/Fieldcollections.php [hunk 4]
```diff
@@ -46,7 +46,7 @@ class Fieldcollections extends Data implements CustomResourcePersistingInterface
      *
      * @var bool
      */
-    public $lazyLoading;
+    public $lazyLoading = false;
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Fieldcollections.php]: BC Break? see above statement

File: models/DataObject/ClassDefinition/Data/Input.php [hunk 5]
```diff
@@ -103,14 +103,14 @@ class Input extends Data implements
      *
      * @var bool
      */
-    public $unique;
+    public $unique = false;
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Input.php]: BC Break? see above statement

File: bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php (hunk 1 above)
Comment (blankse) [reply bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php]: Do you also mean we can change it to this in Pimcore 11?
```
    protected function saveUserToSession(User $user): void
    {
        if (Authentication::isValidUser($user->getUser())) {
```

The parameter types should not be what has effect, but what types may be passed.

Comment (dvesh3) [reply bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php]: yes. this should require `User` type to be passed on Pimcore 11.

Comment (blankse) [reply bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php]: I changed it: https://github.com/pimcore/pimcore/pull/12862/commits/ddcad9d2c3b53614b169de3d18900c948285172a

File: models/DataObject/ClassDefinition/Data/FieldDefinitionEnrichmentInterface.php (hunk 3 above)
Comment (blankse) [reply models/DataObject/ClassDefinition/Data/FieldDefinitionEnrichmentInterface.php:32]: Returns always `$this`, so it should be static return type

File: models/DataObject/ClassDefinition/Data/Block.php (hunk 2 above)
Comment (blankse) [reply models/DataObject/ClassDefinition/Data/Block.php]: I don't think so. The PhpDocs says it is a bool, so its a bug to return `null`. I also see no use case to do compare this with `null`.

Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Block.php]: Agree, the example is not correct. However, these phpdocs are not completely right at the moment and that's what we are fixing in this PR. 

Also keeping BC promise in mind, we should not change the property values in bugfix release.

I would simply move these changes to a separate PR for Pimcore 11, where it is allowed to introduce breaking changes. Could you please take care of it?

Comment (blankse) [reply models/DataObject/ClassDefinition/Data/Block.php]: Okay I reverted it and create a PR for the 11.x branch: https://github.com/pimcore/pimcore/pull/12999

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@blankse thanks a lot!

**Pull request merged**